contexts/global-effects.tsx

```diff
@@ -40,33 +40,33 @@ export const GlobalEffectsProvider = ({children}) => {
 
   useEffect(dao.changeNetwork, [
     Service?.active, 
-    Service?.network?.active?.networkAddress, 
-    connectedChain?.matchWithNetworkChain
+    Service?.network?.active?.networkAddress,
+    connectedChain?.matchWithNetworkChain,
+    query?.chain
   ]);
-
   useEffect(dao.changeChain, [
     connectedChain?.matchWithNetworkChain,
     currentUser?.walletAddress
   ]);
 
-  useEffect(repos.loadRepos, [query?.network , Service?.network?.lastVisited]);
+  useEffect(repos.loadRepos, [
+    query?.network,
+    query?.chain,
+    Service?.network?.lastVisited,
```

Comment:
I think that `lastVisited, active and supportedChains` is a bit overkill, no? x)